drivers/rtc/pcf85263/pcf85263.c

```diff
@@ -0,0 +1,251 @@
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/******************************************************************************/
+
+/**
+ * @brief Read device register.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int pcf85263_read(struct pcf85263_dev *dev, uint8_t reg_addr, uint8_t *reg_data)
+{
+	int ret;
+
+	ret = no_os_i2c_write(dev->i2c_desc, &reg_addr, 1, 0);
+	if (ret)
+		return ret;
+
+	return no_os_i2c_read(dev->i2c_desc, reg_data, 1, 1);
+}
+
+/**
+ * @brief Write device register.
+ * @param dev- The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int pcf85263_write(struct pcf85263_dev *dev, uint8_t reg_addr, uint8_t reg_data)
+{
+	uint8_t buff[2] = { 0 };
+
+	buff[0] = reg_addr;
+	buff[1] = reg_data;
+
+	return no_os_i2c_write(dev->i2c_desc, buff, 2, 1);
+}
+
+/**
+ * @brief Update specific register bits.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - Specific bits mask.
+ * @param reg_data - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int pcf85263_update_bits(struct pcf85263_dev *dev, uint8_t reg_addr,
+			 uint8_t mask, uint8_t reg_data)
+{
+	int ret;
+	uint32_t data;
+
+	ret = pcf85263_read(dev, reg_addr, &data);
+	if (ret)
+		return ret;
+
+	data &= ~mask;
+	data |= reg_data;
+
+	return pcf85263_write(dev, reg_addr, data);
+}
+
+/**
+ * @brief Initialize the device.
+ * @param device - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       parameters.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int pcf85263_init(struct pcf85263_dev **device,
+		  struct pcf85263_init_param init_param)
+{
+	struct pcf85263_dev *dev;
+	uint32_t chip_id;
+	int ret;
+
+	dev = (struct pcf85263_dev *)calloc(1, sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	/* I2C Initialization*/
+	ret = no_os_i2c_init(&dev->i2c_desc, init_param.i2c_init);
+	if (ret)
+		goto error_dev;
+
+	dev->battery_en = init_param.battery_en;
+
+	ret = pcf85263_update_bits(dev, PCF85263_REG_BATTERY_SWITCH,
+				   PCF85263_BATTERY_SW_MSK,
+				   no_os_field_prep(PCF85263_BATTERY_SW_MSK, dev->battery_en));
+	if (ret)
+		goto error_i2c;
+
+	*device = dev;
+
+	return 0;
+
+error_i2c:
+	no_os_i2c_remove(dev->i2c_desc);
+error_dev:
+	free(dev);
+
+	return ret;
+}
+
+/**
+ * @brief Set date.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int pcf85263_set_date(struct pcf85263_dev *dev, struct pcf85263_date date)
```

Comment:
missing date parameter in function comment... I think the CI build also detected this.